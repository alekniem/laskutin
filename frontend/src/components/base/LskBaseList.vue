<template>
    <v-list :id="cid" dense>
        <slot />
    </v-list>
</template>

<script>
export default {
    name: "LskBaseList",

    props: {
        cid: { type: String, required: true },
    },
};
</script>
