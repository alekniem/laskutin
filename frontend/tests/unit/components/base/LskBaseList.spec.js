import LskBaseList from "@/components/base/LskBaseList";
import { createWrapper } from "UT/lsk_test_helper";

describe("LskBaseList", () => {
    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseList, {
            propsData: {
                cid: "TEST_CID",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("no content in default slot");
    });

    it("should match snapshot", () => {
        const wrapper = createWrapper(LskBaseList, {
            propsData: {
                cid: "TEST_CID",
            },
            slots: {
                default: "<div>DEFAULT</div>",
            },
        });
        expect(wrapper.html()).toMatchSnapshot("content in default slot");
    });
});
