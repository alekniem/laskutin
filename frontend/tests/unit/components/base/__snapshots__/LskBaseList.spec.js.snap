// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LskBaseList should match snapshot: content in default slot 1`] = `
<div role="list"
     id="TEST_CID"
     class="v-list v-sheet theme--light v-list--dense"
>
  <div>
    DEFAULT
  </div>
</div>
`;

exports[`LskBaseList should match snapshot: no content in default slot 1`] = `
<div role="list"
     id="TEST_CID"
     class="v-list v-sheet theme--light v-list--dense"
>
</div>
`;
